fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java

```diff
@@ -174,6 +179,60 @@ public FrontendServiceImpl(ExecuteEnv exeEnv) {
         this.exeEnv = exeEnv;
     }
 
+    @Override
+    public TStatus confirmUnusedRemoteFiles(TConfirmUnusedRemoteFilesRequest request) throws TException {
+        TStatus res = new TStatus(TStatusCode.OK);
+        if (!request.isSetCooldownMetaId()) {
+            res.setStatusCode(TStatusCode.INVALID_ARGUMENT);
+            res.addToErrorMsgs("cooldown_meta_id is null");
+            return res;
+        }
+        TabletMeta tabletMeta = Env.getCurrentEnv().getTabletInvertedIndex().getTabletMeta(request.tablet_id);
+        if (tabletMeta == null) {
+            res.setStatusCode(TStatusCode.NOT_FOUND);
+            res.addToErrorMsgs("tablet not found");
+            return res;
+        }
+        Tablet tablet;
+        try {
+            OlapTable table = (OlapTable) Env.getCurrentInternalCatalog().getDbNullable(tabletMeta.getDbId())
+                    .getTable(tabletMeta.getTableId())
+                    .get();
+            table.readLock();
+            try {
+                tablet = table.getPartition(tabletMeta.getPartitionId()).getIndex(tabletMeta.getIndexId())
+                        .getTablet(request.tablet_id);
+            } finally {
+                table.readUnlock();
+            }
+        } catch (RuntimeException e) {
+            res.setStatusCode(TStatusCode.NOT_FOUND);
+            res.addToErrorMsgs("tablet not found");
+            return res;
+        }
+        // check cooldownReplicaId
+        long cooldownReplicaId = tablet.getCooldownConf().first;
+        if (cooldownReplicaId != request.cooldown_replica_id) {
+            res.setStatusCode(TStatusCode.ABORTED);
+            res.addToErrorMsgs(
+                    "cooldown replica id not match(" + cooldownReplicaId + " vs " + request.cooldown_replica_id + ')');
+            return res;
+        }
+        // check cooldownMetaId of all replicas are the same
+        List<Replica> replicas = Env.getCurrentEnv().getTabletInvertedIndex().getReplicas(request.tablet_id);
+        for (Replica replica : replicas) {
```

Comment:
cooldown_replica_id need to be equal, so that if meta_id doesn't match, max rowset version checked is also useful.